iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/analyzer/ExpressionAnalyzer.java

```diff
@@ -1150,6 +1151,11 @@ protected Type visitNode(Node node, StackableAstVisitorContext<Context> context)
           String.format("not yet implemented: %s", node.getClass().getName()));
     }
 
+    @Override
+    protected Type visitColumns(Columns node, StackableAstVisitorContext<Context> context) {
+      throw new IllegalStateException("Columns shouldn't appear in Expression Analyze");
```

Comment:
SemanticException (701 error code) is better.
![image](https://github.com/user-attachments/assets/6a6ab772-5dc0-4259-b2c9-10bda91fed99)